app/frontend/app/models/story.js

```diff
@@ -30,6 +31,8 @@ export default Model.extend(ModelCurrentUser, {
   }.property('poster.id', 'user.id'),
 
   canDeleteStory: function() {
-    return (!this.get('isNew')) && (this.get('belongsToUser') || this.get('currentUser.isAdmin'));
-  }.property('isNew', 'belongsToUser', 'currentUser.isAdmin'),
+    var groupMember = this.get('group') && this.get('group').isMember(this.get('currentUser'));
+    return (!this.get('isNew')) && (this.get('belongsToUser') || this.get('currentUser.isAdmin') ||
+      (groupMember && (groupMember.get('isAdmin') || groupMember.get('isMod'))));
+  }.property('isNew', 'belongsToUser', 'currentUser.isAdmin', 'group'),
```

Comment:
My soul aches from this conditional.  Can we split it onto multiple lines and also like `if this.get('currentUser.isAdmin') return true;` to emphasize the fact that admin is short-circuited true?